packages/worker/src/worker.ts

```diff
@@ -924,9 +928,17 @@ export class Worker {
         ),
         { defaultValue: undefined }
       );
-    } finally {
+      // Only shutdown the native worker if we completed without an error.
```

Comment:
> Only shutdown the native worker if we completed without an error.

I may be misunderstanding the comment, but native worker complete shutdown is called regardless of error